### PR TITLE
Cleanup for close vectors functionality

### PR DESCRIPTION
**Pull request by thofma**
Here are the changes implemented after discussions with @simonbrandhorst:

- Only user facing function is `close_vectors(::ZLat, ::Vector, ::Any)`. Return values are the vectors and the distance.
- Add a `filter` keyword.
- Add tests testing the result.
- Remove quadratic triple code.
- Keep `closest_vectors` functions for now (backwards compatibility).

### CODE REVIEW

File: src/QuadForm/CloseVectors.jl [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+end
+
+function _close_vectors(L::ZLat, v::Vector{fmpq}, upperbound::fmpq;
+                                equal::Bool=false, sorting::Bool=false,
+                                check=true, filter = nothing) where T <: RingElem
+  epsilon = QQ(1//10)   # some number > 0, not sure how it influences performance
+  d = length(v)
+  G1 = gram_matrix(rational_span(L))
+  if check
+    @req is_definite(L) == true && G1[1, 1] > 0 "Zlattice must be positive definite"
```
Comment (simonbrandhorst) [inline src/QuadForm/CloseVectors.jl:47]:
```suggestion
    @req is_positive_definite(L)  "Zlattice must be positive definite"
```

File: src/QuadForm/CloseVectors.jl [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  V = rational_span(L)
+  for a in sv
+    _a, _l = a
+    al = _a[end]
+    if iszero(al)
+      continue
+    end
+    @assert al == 1 || al == -1
+
+    x = fmpz[_a[i] for i in 1:d]
```
Comment (simonbrandhorst) [inline src/QuadForm/CloseVectors.jl:86]: What is your reason for this conversion? We are not doing it for `short_vectors` so why here?

**Review by simonbrandhorst — APPROVED** [review]

File: src/QuadForm/CloseVectors.jl (hunk 2 above)
Comment (thofma) [reply src/QuadForm/CloseVectors.jl:86]: It was there before and I guess you had a reason. I will adjust it.

File: src/QuadForm/CloseVectors.jl (hunk 1 above)
Comment (thofma) [reply src/QuadForm/CloseVectors.jl]: Thanks, I forgot that we had that.

**Comment by thofma** [comment]
I fix the filter in a separate PR.

**Pull request merged**